testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2202,6 +2218,226 @@ public void testTrinoAnalyzeWithNonLowercaseColumnName()
+        String tableLocation = getTableLocation(trinoTableName, true);
+        // Drop table from hive metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(baseTableName);
+
+        String baseTableNameNew = "test_register_table_with_different_table_name_new_" + storageFormat + "_" + randomTableSuffix();
+        String trinoTableNameNew = trinoTableName(baseTableNameNew);
+        String sparkTableNameNew = sparkTableName(baseTableNameNew);
+
+        onTrino().executeQuery(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableNameNew, tableLocation));
+        onSpark().executeQuery(format("INSERT INTO %s values(3, 'POLAND', true)", sparkTableNameNew));
+        List<Row> expectedNew = List.of(row(1, "INDIA", true), row(2, "USA", false), row(3, "POLAND", true));
+
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s", trinoTableNameNew))).containsOnly(expectedNew);
+        onTrino().executeQuery(format("DROP TABLE %s", trinoTableNameNew));
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testRegisterTableWithDroppedTable()
+    {
+        StorageFormat storageFormat = StorageFormat.ORC;
+        String baseTableName = "test_register_table_with_dropped_table_" + storageFormat + "_" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s')", trinoTableName, storageFormat));
+        onSpark().executeQuery(format("INSERT INTO %s values(1, 'INDIA', true)", sparkTableName));
+        onTrino().executeQuery(format("INSERT INTO %s values(2, 'USA', false)", trinoTableName));
+        List<Row> expected = List.of(row(1, "INDIA", true), row(2, "USA", false));
+
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s", trinoTableName))).containsOnly(expected);
+
+        String tableLocation = getTableLocation(trinoTableName);
+        // Drop table to verify register_table call fails when no metadata can be found (table doesn't exist)
+        onTrino().executeQuery(format("DROP TABLE %s", trinoTableName));
+
+        String baseTableNameNew = "test_register_table_with_invalid_table_location_and_metadata_file_new_" + storageFormat + "_" + randomTableSuffix();
+        assertQueryFailure(() -> onTrino().executeQuery(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableNameNew, tableLocation)))
+                .hasMessageMatching(".*No versioned metadata file exists at location.*");
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testRegisterTableWithNoMetadataFile()
+            throws IOException, TException
+    {
+        // TODO find the way to use filesystem api, delete metadata and then enable the test
```

Comment:
```
    @Inject
    private HdfsClient hdfsClient;
```

and

https://github.com/trinodb/trino/blob/6e29d60fd99370e197739bf74e00cb20725bb4d4/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java#L2019-L2026

Reply:
A bit of context behind `hdfsClient` 

https://github.com/trinodb/trino/blob/6e29d60fd99370e197739bf74e00cb20725bb4d4/testing/trino-product-tests/src/main/resources/tempto-configuration.yaml#L1-L4

https://hadoop.apache.org/docs/r1.0.4/webhdfs.html